_posts/2024-11-05-lifetime-event-loop.md

```diff
@@ -0,0 +1,89 @@
+---
+title: "Lifetime Event Loop"
+author: straight-shoota,ysbaddaden
+summary: ""
+categories: technical
+tags: [feature, eventloop, concurrency]
+---
+
+A core component of Crystal's concurrency model is the **event loop**. It
+integrates asynchronous operations into the runtime, enabling other fibers to
+run while one fiber waits to read data on a socket, for example.
+
+We're changing how the event loop operates under the hood on Unix systems in [#14996].
+
+The new implementation integrates the Crystal event loop directly with system
+selectors, [`epoll`](https://linux.die.net/man/7/epoll) (Linux, Android) and
+[`kqueue`](https://man.freebsd.org/cgi/man.cgi?kqueue) (BSDs, macOS) instead of
+going through [`libevent`](https://libevent.org/).
+
+In the process we're changing that file descriptors now stay subscribed to the
+event loop for their entire lifetime, instead of being added and removed on
+every single operation. This reduces overhead and improves performance.
```

Comment:
Maybe something like:

> Not short of removing an external dependency, and taking control over a core feature, this also changes how file descriptors are treated: instead of being added and removed on every blocking IO operation, the file descriptors are now added once and kept for their full lifetime, which is how `epoll` and `kqueue` have been designed.